enhancements/installer/aws-custom-edge-machineset-local-zones.md

```diff
@@ -0,0 +1,933 @@
+- Check if the nodes will be created with defaults (`gp2` and instances)
+- Check if the instances on LAX are `m5.xlarge`
+- Check if the instance on LAS is `c5d.2xlarge`
+- Remove the LAS `us-west-2-las-1a` MachineSet
+- Remove the LAX `us-west-2-lax-1b` MachineSet
+- Scale down the LAX `us-west-2-lax-1a` MachineSet
+- Check if the Machines will be removed correctly
+- Scale up the LAX MachineSet `us-west-2-lax-1a`
+- Check if the new instance was created with gp3 on the zone `us-west-2-lax-1a`
+
+#### Phase 2 test
+
+We will add a new (?) step to install the cluster with support of `edge` machine pools in the
+zones of LAX (`us-west-2-lax-1a`, `us-west-2-lax-1b`) and LAS (`us-west-2-las-1a`).
+
+The test will:
+
+- Add a ipi-conf step to set the `edge` machine pool setting the `compute[].platform.aws.zones`
+  to the LAX and LAS zones (`us-west-2-lax-1a`, `us-west-2-lax-1b`, `us-west-2-las-1a`)
+- Check if the nodes will be created with defaults (`gp2` and instances)
+- Check if the instances on LAX are `m5.xlarge`
+- Check if the instance on LAS is `c5d.2xlarge`
+- Remove the LAS `us-west-2-las-1a` MachineSet
+- Remove the LAX `us-west-2-lax-1b` MachineSet
+- Scale down the LAX `us-west-2-lax-1a` MachineSet
+- Check if the Machines will be removed correctly
+- Scale up the LAX MachineSet `us-west-2-lax-1a`
+- Check if the new instance was created with `gp3` on the zone `us-west-2-lax-1a`
+- Deploy one sample application with node selector to `zone_group=us-west-2-lax-1`
+- Check if the application has been created on the node created by the MachineSet `us-west-2-lax-1a`
+
+#### E2E test
+
+TBD
+
+### Graduation Criteria
+#### Dev Preview -> Tech Preview
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+N/A. The resource creation, Local Zones subnets, and MachineSets in the edge
+machine pool are available only in day zero operations. All the further
+lifecycle will be done as a normal machine.
+
+### Version Skew Strategy
+
+N/A. The same as `Upgrade / Downgrade Strategy`.
+
+### Operational Aspects of API Extensions
+
+N/A.
+
+#### Failure Modes
+
+N/A.
+
+#### Support Procedures
+
+N/A.
+
+## Implementation History
+
+Currently, a few research documents were created to understand how the edge networks
+of AWS can be used on OKD/OpenShift, most of them were focused on extending current clusters
+using MachineSets when installing a cluster in existing networks.
+
+The first research was focused on Day-2 operations, how to extend current IPI deployment
+to Local Zones subnets, created separately, and how to use the workloads on it, since the
+openshift router supports only network load balancers (NLB/CLB), which is not supported
+on Local Zones.
+
+## Alternatives
+
+- Cluster admins can create subnets and the MachineSet manifest at the day zero operations
+  using the installer option `create manifest`, then trigger the cluster creation.
```

Comment:
Probably worth explicitly explaining why these alternatives were not chosen.  Something like:

> But manually creating subnets and MachineSets is tedious and error-prone.  And there are no in-cluster operators to assist with these changes post-install.  So we're sticking the helper in the installer to deliver user convenience with minimal developer effort.

or whatever the motivation for avoiding the alternatives actually is.

Reply:
[Looks like][1] the `subnet` pointer is just about allowing HostedCluster lookups, not about day-2 creation when adding a NodePool.  But that's today, and I still think it would be nice to have active controllers for this infra at some point.  But folks may disagree, and even if they agree with me, that active-controller point may be so far out in the distant future that it's not worth planning around today.

[1]: https://github.com/openshift/hypershift/blob/4073d66298251f2c7bb790c5be4afbec6af4dc28/cmd/nodepool/aws/create.go#L53-L59

Reply:
> Probably worth explicitly explaining why these alternatives were not chosen. Something like:
> 
> > But manually creating subnets and MachineSets is tedious and error-prone.  And there are no in-cluster operators to assist with these changes post-install.  So we're sticking the helper in the installer to deliver user convenience with minimal developer effort.
> 
> or whatever the motivation for avoiding the alternatives actually is.

Good point, thanks for raising it. I will update you with that suggestions. In the Motivation section, I put the long-list of tasks to do, pointing this with your suggestions *sgtm*.